### DISCUSSION TITLE
Not loading EXP correctly.

### DISCUSSION THREAD

**Discussion by DasLixou** [Minestom/Minestom#876] [Help] [answered]
Hello! I set my Level & experience in the PlayerLoginEvent with this code:
```java
player.setLevel(2022);
player.setExp(1);
```

but when i join the exp and level are still 0;

BUT: when i use my nicksystem and change my player with:

```java
player.setSkin(PlayerSkin.fromUsername(randomUserName));
player.setDisplayName(Component.text(randomUserName));
```

then it shows the exp and level correctly. 

How can i "update" the player so it works directly?

**Comment by Eoghanmc22** [comment, marked as answer]
needs to be in PlayerSpawnEvent. you can check firstSpawn if needed. the player (clientside) does not exist yet in the login event

**Reply by DasLixou** [reply]
Thanks :D
